String/BoyerMoore.js

```diff
@@ -0,0 +1,58 @@
+ /*
+ *
+ *
+ *Implementation of the Boyer-Moore String Search Algo.
+ *The Boyer–Moore string search algorithm allows linear time in
+ *      search by skipping
+ *      indices when searching inside a string for a pattern.
+ *
+ *
+ *
+ *
+ **/
+const buildBadMatchTable = (str) => {
+  let tableObj = {};
+  let strLength = str.length;
+  for (var i = 0; i < strLength - 1; i++) {
+      tableObj[str[i]] = strLength - 1 - i;
+  }
+  if (tableObj[str[strLength-1]] == undefined) {
+      tableObj[str[strLength-1]] = strLength;
+  }
+  return tableObj;
+}
+
+
+let boyerMoore = (str, pattern)=> {
+  var badMatchTable = buildBadMatchTable(pattern),
+  offset = 0,
+  patternLastIndex = pattern.length - 1,
+  scanIndex = patternLastIndex,
+  maxOffset = str.length - pattern.length;
+  // if the offset is bigger than maxOffset, cannot be found
+  while (offset <= maxOffset) {
```

Comment:
Please fix your indentation.

Reply:
I just completed the fixes.

Reply:
You still didn't fix the indentation.

Reply:
The last commit, this 40bd41c3be55e44b8cd1d704aa56913bce24c254 did.
It passed the code style check & the lgtm check